ServerCore/Pages/Events/FastestSolves.cshtml.cs

```diff
@@ -50,23 +65,26 @@ public async Task OnGetAsync(SortOrder? sort, PuzzleStateFilter? stateFilter)
                 .Select(g => new {
                     Puzzle = g.Key,
                     SolveCount = g.Count(),
-                    Fastest = g.OrderBy(s => s.SolvedTime - s.UnlockedTime).Take(3).Select(s => new { s.Team.ID, Time = s.SolvedTime - s.UnlockedTime}),
-                    IsSolvedByUserTeam = g.Where(state => state.TeamID == this.CurrentTeam.ID).Any()
+                    Fastest = g.OrderBy(s => s.SolvedTime - s.UnlockedTime).Take(3).Select(s => new { s.Team.ID, Time = s.SolvedTime - s.UnlockedTime }),
+                    IsSolvedByUserTeam = g.Where(s => s.Team == this.CurrentTeam).Any()
                 })
                 .OrderByDescending(p => p.SolveCount).ThenBy(p => p.Puzzle.Name)
                 .ToListAsync();
 
             var unlockedData = new HashSet<int>(
-                    await PuzzleStateHelper.GetSparseQuery(_context, this.Event, null, null)
-                    .Where(s => s.TeamID != this.CurrentTeam.ID && s.UnlockedTime != null)
-                    .Select(s => s.PuzzleID)
-                    .ToListAsync());
+                PuzzleStateHelper.GetSparseQuery(_context, this.Event, null, null)
```

Comment:
For perf we should keep the awaits and the previous where clause. How about we only fill this in when this.CurrentTeam is non-null?